test/jdk/sun/security/provider/acvp/Launcher.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import jdk.test.lib.json.JSONValue;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.Provider;
+import java.security.Security;
+
+/*
+ * @test
+ * @bug 8342442
+ * @library /test/lib
+ */
+public class Launcher {
+
+    private static final String ONLY_ALG
+            = System.getProperty("acvp.test.alg");
+
+    private static final Provider PROVIDER;
+
+    static {
+        var provProp = System.getProperty("acvp.test.provider");
+        PROVIDER = provProp != null
+                ? Security.getProvider(provProp)
+                : null;
+    }
+
+    public static void main(String[] args) throws Exception {
+
```

Comment:
Use of properties to configure the test is a bit puzzling since the test is a main(String[] args) program.
It would more convenient to run it from the command line (where it would be run manually) to use normal arguments.

It would be better to describe the properties, if retained, as "test properties", not system properties since they only apply to the test and not the implementation. A "test." prefix on the property names would also make it clearer.

Including sample `* @run main/manual ...` lines might show more easily how they would be used.

Reply:
We will put test data files into the `data` directory in future commits, and then the test can run as an automatic test without any test properties. The test can also be launched directly with the `jtreg` command while development new algorithms, and the test properties would be useful. I didn't meant to make it a manual test.

I'll rename the property names. Thanks.